agent/src/workload/workload_control_loop.rs

```diff
@@ -64,28 +66,190 @@ impl RetryCounter {
+
+                            log::debug!("Update workload complete");
+                        }
+                        // [impl->swdd~agent-workload-control-loop-executes-retry~1]
+                        Some(WorkloadCommand::Retry(instance_name)) => {
+                            log::debug!("Received WorkloadCommand::Retry.");
+
+                            control_loop_state = Self::retry_create_workload_on_runtime(
+                                control_loop_state,
+                                *instance_name,
+                            )
+                            .await;
+                        }
+                        // [impl->swdd~agent-workload-control-loop-executes-create~2]
+                        Some(WorkloadCommand::Create) => {
+                            log::debug!("Received WorkloadCommand::Create.");
+
+                            control_loop_state = Self::create_workload_on_runtime(
+                                control_loop_state,
+                                Self::send_retry_for_workload,
+                            )
+                            .await;
+                        }
+                        // [impl->swdd~agent-workload-control-loop-executes-resume~1]
+                        Some(WorkloadCommand::Resume) => {
+                            log::debug!("Received WorkloadCommand::Resume.");
+                            control_loop_state = Self::resume_workload_on_runtime(control_loop_state).await;
+                        }
+                        _ => {
+                            log::warn!(
+                                "Could not wait for internal stop command for workload '{}'.",
+                                control_loop_state.instance_name().workload_name(),
+                            );
+                            return;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    async fn send_workload_state_to_agent(
+        workload_state_sender: &WorkloadStateSender,
+        instance_name: &WorkloadInstanceName,
+        execution_state: ExecutionState,
+    ) {
+        workload_state_sender
+            .report_workload_execution_state(instance_name, execution_state)
+            .await;
+    }
+
+    // [impl->swdd~workload-control-loop-handles-workload-restarts~1]
+    async fn restart_if_workload_state_matches_restart_policy<WorkloadId, StChecker>(
+        mut control_loop_state: ControlLoopState<WorkloadId, StChecker>,
+        new_workload_state: WorkloadState,
+    ) -> ControlLoopState<WorkloadId, StChecker>
+    where
+        WorkloadId: ToString + Send + Sync + 'static,
+        StChecker: StateChecker<WorkloadId> + Send + Sync + 'static,
+    {
+        if Self::is_restart_required(&control_loop_state.workload_spec, &new_workload_state) {
+            control_loop_state = Self::restart_workload_on_runtime(control_loop_state).await;
+        }
+        control_loop_state
+    }
```

Comment:
I don't get why we do we need the function here. I proposed to put the if inside the select directly. Now we have a function with 3 lines of code that has an `if` in the name and performs an if.

Reply:
Removed